runtime/src/main/java/org/corfudb/runtime/collections/StreamListener.java

```diff
@@ -6,6 +6,11 @@
  * Created by hisundar on 2019-10-18
  */
 public interface StreamListener {
+
+    default void onNextEntry(CorfuStreamEntries results) {
```

Comment:
What's the purpose of `onNextEntry()` given that we already have `onNext()`?

Reply:
If you look at StreamListenerResumePolicy, we need onNext to set the lastProcessedTimestamp on each onNext call, so onNextEntry ensures that happens after every onNext (leaving that to the listener implementation is error prone as they might not do it) so that's the purpose of onNextEntry

Reply:
It is really to be able to support the 'resume' portion of these new patterns. On every onNext call we need to keep track of the last processed entry, so this is just a wrapper over the onNext callback, which keeps track of this timestamp, right after the user executes its onNext logic.